pkg/controller/network/resourcerecordset/controller_test.go

```diff
@@ -0,0 +1,365 @@
+			MockUpdate: test.NewMockUpdateFn(nil),
+		},
+	}
+
+	os.Exit(m.Run())
+}
+
+func Test_Observe(t *testing.T) {
+	g := gomega.NewGomegaWithT(t)
+
+	name := "x.x.x."
+	mockManaged := v1alpha3.ResourceRecordSet{}
+	meta.SetExternalName(&mockManaged, name)
+
+	incorrectName := "p.p.p."
+	mockManagedRRNotPresent := v1alpha3.ResourceRecordSet{
+		Spec: v1alpha3.ResourceRecordSetSpec{
+			ForProvider: v1alpha3.ResourceRecordSetParameters{
+				Name: &incorrectName,
+			},
+		},
+	}
+	meta.SetExternalName(&mockManagedRRNotPresent, name)
+
+	mockManagedRRPresent := v1alpha3.ResourceRecordSet{
+		Spec: v1alpha3.ResourceRecordSetSpec{
+			ForProvider: v1alpha3.ResourceRecordSetParameters{
+				Name: &name,
+			},
+		},
+	}
+	meta.SetExternalName(&mockManagedRRPresent, name)
+
+	mockExternal := &route53.ResourceRecordSet{
+		Name: aws.String(name),
+	}
+
+	var mockClientErr error
+	var itemsList []route53.ResourceRecordSet
+
+	mockClient.MockChangeResourceRecordSetsRequest = func(input *route53.ChangeResourceRecordSetsInput) route53.ChangeResourceRecordSetsRequest {
+		return route53.ChangeResourceRecordSetsRequest{
+			Request: &aws.Request{
+				HTTPRequest: &http.Request{},
+				Data:        &route53.ChangeResourceRecordSetsOutput{},
+				Error:       mockClientErr,
+			},
+		}
+	}
+
+	mockClient.MockListResourceRecordSetsRequest = func(input *route53.ListResourceRecordSetsInput) route53.ListResourceRecordSetsRequest {
+		return route53.ListResourceRecordSetsRequest{
+			Request: &aws.Request{
+				HTTPRequest: &http.Request{},
+				Data: &route53.ListResourceRecordSetsOutput{
+					ResourceRecordSets: itemsList,
+				},
+				Error: mockClientErr,
+			},
+		}
+	}
+
+	for _, tc := range []struct {
+		description               string
+		managedObj                resource.Managed
+		itemsReturned             []route53.ResourceRecordSet
+		clientErr                 error
+		expectedErrNil            bool
+		expectedResourceExist     bool
+		expectedResourceAvailable bool
+	}{
+		{
+			"upexpected managed resource should return error",
+			unexpectedItem,
+			nil,
+			nil,
+			false,
+			false,
+			false,
+		},
+		{
+			"if item's identifier is not yet set, returns expected",
+			&v1alpha3.ResourceRecordSet{},
+			nil,
+			nil,
+			true,
+			false,
+			false,
+		},
+		{
+			"if external resource doesn't exist, it should return expected",
+			&mockManagedRRNotPresent,
+			[]route53.ResourceRecordSet{*mockExternal},
+			nil,
+			true,
+			false,
+			true,
+		},
+		{
+			"If external resource fails, it should return error",
+			mockManaged.DeepCopy(),
+			nil,
+			errors.New("Error in API call"),
+			false,
+			false,
+			false,
+		},
+	} {
+		mockClientErr = tc.clientErr
+		itemsList = tc.itemsReturned
+		res, err := mockExternalClient.Observe(context.Background(), tc.managedObj)
+
+		g.Expect(err == nil).To(gomega.Equal(tc.expectedErrNil), tc.description)
+		g.Expect(res.ResourceExists).To(gomega.Equal(tc.expectedResourceExist), tc.description)
```

Comment:
Similar to above, I'd like to avoid using gomega in favour of our table driven testing convention.

Reply:
The requested test has been rewritten with the suggested table driven style.